### PR TITLE
return not-found errors properly from fake listers

### PR DESCRIPTION
**Pull request by utam0k**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Implementing https://github.com/kubernetes/kubernetes/pull/121508, I found `apierrors.IsNotFound` doesn't work well. Because the original error was returned by the fake lister,` apierrors.IsNotFound` did not work properly. This was not a good way to test, so it has been fixed.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by utam0k** [comment]
If it is acceptable, I'll try to fix others.

**Comment by utam0k** [comment]
@sanposhiho Probably, this PR relates to your PR https://github.com/kubernetes/kubernetes/pull/121952.

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

File: pkg/scheduler/testing/framework/fake_listers.go [hunk 1]
```diff
@@ -195,7 +195,7 @@ func (f *persistentVolumeClaimNamespaceLister) Get(name string) (*v1.PersistentV
 			return pvc, nil
 		}
 	}
-	return nil, fmt.Errorf("persistentvolumeclaim %q not found", name)
+	return nil, errors.NewNotFound(v1.Resource("persistentvolumeclaim"), name)
```
Comment (sanposhiho) [inline pkg/scheduler/testing/framework/fake_listers.go:198]:
```suggestion
	return nil, errors.NewNotFound(v1.Resource("persistentvolumeclaims"), name)
```

must be plural

**Comment by sanposhiho** [comment]
> If it is acceptable, I'll try to fix others.

@utam0k could you please

**Comment by utam0k** [comment]
> > If it is acceptable, I'll try to fix others.
> 
> @utam0k could you please

Sure. I will do it after merging this PR.

**Comment by sanposhiho** [comment]
Why not in this PR? I'd rather put all fixes in this PR as it's just a fix in testing stuff.

**Comment by utam0k** [comment]
> Why not in this PR? I'd rather put all fixes in this PR as it's just a fix in testing stuff.

It was just my mood, so I will do that 👍

File: pkg/scheduler/testing/framework/fake_listers.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (f ControllerLister) GetPodControllers(pod *v1.Pod) (controllers []*v1.Repl
 		}
 	}
 	if len(controllers) == 0 {
-		err = fmt.Errorf("could not find Replication Controller for pod %s in namespace %s with labels: %v", pod.Name, pod.Namespace, pod.Labels)
+		err = errors.NewNotFound(v1.Resource("replicationcontrollers"), pod.Name)
```
Comment (sanposhiho) [inline pkg/scheduler/testing/framework/fake_listers.go:97]: Apparently, GetXXXs (plural) don't return not-found errors.
e.g., https://github.com/kubernetes/client-go/blob/84a6fe7e4032ae1b8bc03b5208e771c5f7103549/listers/core/v1/replicationcontroller_expansion.go#L62

Comment (utam0k) [reply pkg/scheduler/testing/framework/fake_listers.go:97]: This was just the place to get lost. Thanks for telling me about it. I've learnt 🙏

Comment (utam0k) [reply pkg/scheduler/testing/framework/fake_listers.go]: Fixed 74e076d724aa195677522c623b156a3dff04a8bf

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve
/retitle return not-found errors properly from fake listeres

**Comment by sanposhiho** [comment]
@utam0k Please squash

**Comment by sanposhiho** [comment]
@Huang-Wei Can we get `/milestone v1.29` for it? 
It contains only test-related changes and is blocking some other PRs.

/remove-lgtm
for squash commits.

**Comment by utam0k** [comment]
> @utam0k Please squash

Sure! I have done it.

**Comment by sanposhiho** [comment]
/retitle return not-found errors properly from fake listers

**Comment by utam0k** [comment]
I will check the CI 👀

**Comment by utam0k** [comment]
@sanposhiho Would you review this agiain because of adding the commits to pass the CI 🙏

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

cc @Huang-Wei for `/milestone v1.29` (if it sounds ok to give milestone

**Comment by utam0k** [comment]
@sanposhiho @Huang-Wei What's the status of this issue? Can I help you something? I'm waiting for this PR to continue with other PRs!

**Comment by utam0k** [comment]
@sanposhiho  I see. Thanks for your explanation 🙏

**Pull request merged**